nuget.exe 5.5 fails to restore with MSBuild 14 or below

## Details about Problem

NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe): nuget.exe

NuGet version (x.x.x.xxx): 5.5.1

dotnet.exe --version (if appropriate):

VS version (if appropriate):

OS version (i.e. win10 v1607 (14393.321)):

Worked before? If so, with which NuGet version: 5.4.X

## Detailed repro steps so we can see the same problem

1. `nuget.exe restore -MSBuildVersion 14.0`

Fails with the error:
```
C:\Users\jeffkl\AppData\Local\Temp\NuGetScratch\npvhljmi.kme.nugetrestore.targets(435,34):
error MSB4092: An unexpected token ")" was found at character position 28 in condition
"@(PackageReference->Count()) > 0". [D:\AspNetWithNet35\AspNetWithNet35\AspNetWithNet35.csproj]
```

This was introduced by my change: https://github.com/NuGet/NuGet.Client/commit/1ce56a3cd779ec4ebae39a1b0ce3f7ccc3310584#diff-3b0018f61173d0adf1ba163a23821381R434
https://github.com/NuGet/NuGet.Client/pull/3135

It turns out that the item function `Count()` was only added in MSBuild 14.0 Update 3.  So when using an older version of MSBuild, we can't use this logic.

--- a/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets
+++ b/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets
@@ -443,7 +443,16 @@ Copyright (c) .NET Foundation. All rights reserved.
   <Target Name="_GetRestoreProjectStyle"
     DependsOnTargets="_GetProjectJsonPath;CollectPackageReferences"
     Returns="$(RestoreProjectStyle);$(PackageReferenceCompatibleProjectStyle)">
-    <PropertyGroup>
+
+    <!--
+      Older versions of MSBuild do not support the Count() item function which is an optimization.  Expanding the
+      entire item list into a semicolon delimited string is slower but older versions of MSBuild don't support it so
+      use the older logic if necessary
+    -->
+    <PropertyGroup Condition="'$(MSBuildAssemblyVersion)' &lt; '15.0'">
+      <_HasPackageReferenceItems Condition="'@(PackageReference)' != ''">true</_HasPackageReferenceItems>
+    </PropertyGroup>
+    <PropertyGroup Condition="'$(MSBuildAssemblyVersion)' &gt;= '15.0'">
       <_HasPackageReferenceItems Condition="@(PackageReference->Count()) > 0">true</_HasPackageReferenceItems>
     </PropertyGroup>
 

